src/manifest.c

```diff
@@ -1391,3 +1476,73 @@ struct file *search_file_in_manifest(struct manifest *manifest, const char *file
 	}
 	return NULL;
 }
+
+/* Ideally have the manifest sorted already by this point */
+struct file **manifest_files_to_array(struct manifest *manifest)
+{
+	struct file **array;
+	struct list *iter;
+	struct file *file;
+	int i = 0;
+	int numfiles = manifest->filecount;
+
+	if (!manifest) {
+		fprintf(stderr, "Manifest is NULL!\n");
+		return NULL;
+	}
+
+	array = malloc(sizeof(struct file *) * numfiles);
+
+	iter = list_head(manifest->files);
+	while (iter) {
+		file = iter->data;
+		iter = iter->next;
+		array[i] = file;
+		i++;
+	}
+	return array;
+}
+
+void print_manifest_array(struct file **array, int filecount)
+{
+	for (int i = 0; i < filecount; i++) {
+		printf("[%d]/%d %s\n", i, filecount, array[i]->filename);
+	}
+	printf("SIZE: %d\n", filecount);
+}
+
+void free_manifest_array(struct file **array)
+{
+	free(array);
+}
+
+static int cmpnames(const void *a, const void *b)
+{
+	return strcmp((*(struct file **)a)->filename, (*(struct file **)b)->filename);
+}
+
+int enforce_compliant_manifest(struct file **a, struct file **b, int searchsize, int size)
+{
+	struct file **found;
+	int ret = 0;
+
+	qsort(a, searchsize, sizeof(struct file *), cmpnames);
+	qsort(b, size, sizeof(struct file *), cmpnames);
+	printf("Checking manifest uniqueness...\n");
+	for (int i = 0; i < searchsize; i++) {
+		found = bsearch(a[i], b, size, sizeof(struct file *), bsearch_file_helper);
+		if (found) {
```

Comment:
Do the manifests only have existing files in them, and not for example deleted files? In other words do we have a case where the current manifest has a file, e.g. /etc/oldissue marked as deleted but our new package wants to create this file.

Reply:
If there's any collision the type/action on the file doesn't matter, because it's a conflict between the official manifest & user manifest. This of course can just be ignored, but it's a warning that you're possibly breaking something by overriding it.